app/src/main/java/com/asfoundation/wallet/promotions/voucher/EVoucherDetailsPresenter.kt

```diff
@@ -0,0 +1,34 @@
+package com.asfoundation.wallet.promotions.voucher
+
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.schedulers.Schedulers
+
+class EVoucherDetailsPresenter(
+    private val view: EVoucherDetailsView,
+    private val interactor: EVoucherDetailsInteractor,
+    private val navigator: EVoucherDetailsNavigator,
+    private val data: EVoucherDetailsData
+) {
+
+  private val disposable = CompositeDisposable()
+
+  fun present() {
+    disposable.add(interactor.getSkuButtonModels()
```

Comment:
Two things here:
 - We normally put each subscription into a method for organization purposes (e.g. initializeView(), handleNextClick(), handleCancelClick(), etc. and call it on present()).
 - We don't generally put actions into subscribes but we print the stacktrace on error
 
 Here's an example:
```
   private fun handleNextClickEvents() {
    disposables.add(
        view.onNextClicks()
            .doOnNext { skuButtonModel ->
              navigator.navigateToNextScreen(skuButtonModel.skuId)
            }
            .subscribe({}, { e -> e.printStackTrace() })
    )
  }
```